### PR TITLE
fix: use `ServicesSoftwareVersion`

### LINKED ISSUE #15292: (0.54) `-alpha.x` semvers are not maintained in state
### Description

An `v0.54.0-alpha.1` tag after deserialization from state appeared to be a `v0.54.0` tag.

### Steps to reproduce

As above.

### Additional context

_No response_

### Hedera network

other

### Version

develop

### Operating system

None

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Closes #15292 
 - Replaces `HederaSoftwareVersion` with `ServicesSoftwareVersion`
     * This `SoftwareVersion` implementation keeps all required version information (_semantic-version.properties_ semver plus `hedera.config.version` config version) in a single `SemanticVersion` object.
 - Removes unused `appVersion()` method from `SelfNodeInfo` interface.
 - Fix a typo in `BootstrapUtils` log, which was logging the deserialized state version as the **current** version.
 - Always includes the `HederaSoftwareVersion` class id in permissible set of `SoftwareVersion` ids to support migration from 0.53 pre-consensus event files.

Testing:

- Validate Upgrade passes from 0.54.0 to 0.54.0-alpha.1
- Validated Upgrade passes from 0.53.0 to 0.54.0

### CODE REVIEW

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java [hunk 1]
```diff
@@ -455,23 +453,26 @@ public void onStateInitialized(
...
+        } else if (previousVersion instanceof HederaSoftwareVersion hederaSoftwareVersion) {
+            deserializedVersion = new ServicesSoftwareVersion(
+                    hederaSoftwareVersion.servicesVersion(), hederaSoftwareVersion.configVersion());
         } else {
             if (previousVersion != null) {
                 logger.fatal("Deserialized state not created with Hedera software");
                 throw new IllegalStateException("Deserialized state not created with Hedera software");
             }
         }
+        if (version.compareTo(deserializedVersion) < 0) {
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:469]: Makes sense to keep this outside this case `previousVersion instanceof HederaSoftwareVersion hederaSoftwareVersion` 👍

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java [hunk 2]
```diff
@@ -496,19 +497,20 @@ public void onStateInitialized(
...
     private List<StateChanges.Builder> onMigrate(
             @NonNull final State state,
-            @Nullable final HederaSoftwareVersion deserializedVersion,
+            @Nullable final ServicesSoftwareVersion deserializedVersion,
             @NonNull final InitTrigger trigger,
             @NonNull final Metrics metrics) {
         final var previousVersion = deserializedVersion == null ? null : deserializedVersion.getPbjSemanticVersion();
-        final var currentVersion = version.getPbjSemanticVersion();
-        final var isUpgrade = isSoOrdered(previousVersion, currentVersion);
+        final var isUpgrade = version.compareTo(deserializedVersion) > 0;
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:504]: nit: Seems `previousVersion` is just used in logs

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/version/HederaSoftwareVersion.java [hunk 3]
```diff
@@ -77,17 +75,12 @@ public HederaSoftwareVersion(
         this.servicesVersion = Objects.requireNonNull(servicesVersion, "servicesVersion must not be null");
     }
 
-    /**
```
Comment (imalygin) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/version/HederaSoftwareVersion.java:80]: Given the fact that `ServicesSoftwareVersion` replaces `HederaSoftwareVersion`, should `HederaSoftwareVersion` be deprecated?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/version/HederaSoftwareVersion.java [hunk 4]
```diff
@@ -77,17 +75,12 @@ public HederaSoftwareVersion(
...
-     * @param hapiVersion HAPI version
-     * @param servicesVersion services version
-     */
-    public HederaSoftwareVersion(final SemanticVersion hapiVersion, @NonNull final SemanticVersion servicesVersion) {
-        this.hapiVersion = hapiVersion;
-        this.servicesVersion = Objects.requireNonNull(servicesVersion, "servicesVersion must not be null");
-        if (!StringUtils.isEmpty(servicesVersion.build())) {
-            this.configVersion = Integer.parseInt(servicesVersion.build());
-        }
+    public SemanticVersion servicesVersion() {
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/version/HederaSoftwareVersion.java:78]: We should probably add a comment in this class to say that we need to this class to load any state before 0.53. Can be done in later PR.

**Review by imalygin — APPROVED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]
Makes sense to me.  tyvm @tinker-michaelj!

**Review by lpetrovic05 — APPROVED** [review]
looks good as far as I can tell

**Pull request merged**